datamol/scaffold/_fuzzy.py

```diff
@@ -80,7 +82,8 @@ def fuzzy_scaffolding(
     additional_templates: Optional[List[Mol]] = None,
     ignore_non_ring: bool = False,
     mcs_params: Optional[Dict[Any, Any]] = None,
-) -> Tuple[set, Dict[str, dict], Dict[str, list]]:
+    if_df: bool = True,
```

Comment:
Thanks @Pakman450, can you replace by `return_df` ? otherwise  LGTM

Reply:
Maybe I would suggest to instead totally drop `if_df` and only return dataframe here. I am not too afraid about backward compat here (since it actually makes sense) and I don't see any good reasons for keeping the old returned types.

@maclandrol agree to merge here and tackle the refactoring later. We should likely revisit all the code and break it down in multiple functions.